vestings/merkle_proof.py

```diff
@@ -19,38 +21,33 @@ def combine_and_hash(leaf1: str, leaf2: str):
     return HexBytes(combined_hash).hex()
 
 
-def generate_vestings_tree(elements):
+def generate_vestings_tree(elements: List[HexStr]) -> List[List[HexStr]]:
     count = len(elements)
     previous_level = elements.copy()
-    levels: list(list(str)) = [previous_level]
+    levels: List[List[str]] = [previous_level]
     while count > 1:
-        current_level = []
+        current_level: List[HexStr] = []
         i = 0
         while i < count:
             leaf1 = previous_level[i]
-            leaf2 = EMPTY_HASH if i + 1 >= count else previous_level[i + 1]
-            if int.from_bytes(
-                bytes(HexBytes(leaf1)), byteorder="big", signed=False
-            ) < int.from_bytes(bytes(HexBytes(leaf2)), byteorder="big", signed=False):
-                current_level.append(combine_and_hash(leaf1, leaf2))
-            else:
-                current_level.append(combine_and_hash(leaf2, leaf1))
-            i = i + 2
+            leaf2 = previous_level[i + 1] if i + 1 < count else EMPTY_HASH
+            if HexBytes(leaf1) > HexBytes(leaf2):
+                leaf1, leaf2 = leaf2, leaf1
```

Comment:
Maybe a short comment here would be nice. It is an important line that is easy to miss.

Reply:
Yeah, I'm still refactoring the method as I don't like the while loop

Reply:
Updated